src/Event.cpp

```diff
@@ -25,52 +25,82 @@ void Event::add_value(std::string name, std::string&& arg) {
+
+    if (val == string_values.end()) {
+        std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
+    
+        if ( crash_on_fail == true) {   //if there is an error and crash is set to true crash program
+            exit (-1);
+        } else {
+            return false;
+        }
+    }
+
+    *value = val->second;       //update the value requested
+    return true;
+}
+template<>
+bool Event::get_value(const std::string& name, int* value, bool crash_on_fail) const {
+    const std::unordered_map<std::string, EventValue>::const_iterator val = event_values.find(name);
+
     const auto val = event_values_.find(name);
 
     if (val == event_values_.end()) {
         std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
-        return otherwise;
+
+        if ( crash_on_fail == true) {
+            exit (0);
+        } else {
+            return false;
+        }
     }
 
     if (val->second.type_name != typeid(int).name()) {
         std::cout
                 << "Wrong type int instead of " << val->second.type_name
                 << " in value " << name << " of event " << static_cast<int>(event_type) << "!" << std::endl;
-        return otherwise;
+
+        if ( crash_on_fail == true) {
+            exit (0);
+        } else {
+            return false;
+        }
     }
 
-    return val->second.value.int_type;
+    *value = val->second.value.int_type;
+    return true;
 }
-
 template<>
-float Event::get_value(const std::string& name, float otherwise) const {
-    const auto val = event_values_.find(name);
+bool Event::get_value(const std::string& name, float* value, bool crash_on_fail) const {
+    const auto val = event_values.find(name);
 
     if (val == event_values_.end()) {
         std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
-        return otherwise;
+
+        if ( crash_on_fail == true) {
+            exit (0);
+        } else {
+            return false;
+        }
     }
 
-    if (val->second.type_name != typeid(float).name()) {
+    if (val->second.type_name != typeid(int).name()) {
         std::cout
-                << "Wrong type float instead of " << val->second.type_name
+                << "Wrong type int instead of " << val->second.type_name
                 << " in value " << name << " of event " << static_cast<int>(event_type) << "!" << std::endl;
-        return otherwise;
-    }
 
-    return val->second.value.float_type;
-}
-
-template<>
-std::string Event::get_value(const std::string& name, std::string otherwise) const {
-    const auto val = string_values_.find(name);
-
-    if (val == string_values_.end()) {
-        std::cout << "String Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
-        return otherwise;
+        if ( crash_on_fail == true) {
+            exit (0);
```

Comment:
This should not be exiting with 0. If anything, this should either exit with EXIT_FAILURE or throw an exception (if we want to go down that route).

Reply:
Also missing the error statement. Use std::cerr for printing out error messages.